src/Controllers/FoodVoucherController.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Controllers;
+
+use Engelsystem\Config\Config;
+use Engelsystem\Helpers\Authenticator;
+use Engelsystem\Helpers\Carbon;
+use Engelsystem\Helpers\Translation\Translator;
+use Engelsystem\Helpers\UserVouchers;
+use Engelsystem\Http\Exceptions\HttpForbidden;
+use Engelsystem\Http\Exceptions\HttpNotFound;
+use Engelsystem\Http\Redirector;
+use Engelsystem\Http\Request;
+use Engelsystem\Http\Response;
+use Engelsystem\Models\User\User;
+use Engelsystem\Models\Worklog;
+use ErrorException;
+use GuzzleHttp\Client as GuzzleClient;
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Log\LoggerInterface;
+
+class FoodVoucherController extends BaseController
+{
+    use HasUserNotifications;
+
+    public function __construct(
+        protected Authenticator $auth,
+        protected GuzzleClient $guzzle,
+        protected Config $config,
+        protected LoggerInterface $log,
+        protected Worklog $worklog,
+        protected Redirector $redirect,
+        protected Response $response,
+        protected User $user,
+        protected Translator $translator,
+    ) {
+    }
+
+    private function checkActive(): void
+    {
+        if (!$this->config->get('enable_voucher') || !$this->config->get('enable_force_food')) {
+            throw new HttpNotFound();
+        }
+        if (!$this->userIsForced() && UserVouchers::eligibleVoucherCount($this->auth->user()) == 0) {
+            throw new HttpForbidden();
+        }
+    }
+
+    private function checkConfig(): void
+    {
+        $conf = $this->config->get('food_voucher_api');
+        if (
+            !(
+            $conf
+            && $conf['info_url']
+            && $conf['auth_token']
+            && $conf['post_url']
+            )
+        ) {
+            throw new HttpNotFound();
+        }
+    }
+
+    private function getAuthToken(): string
+    {
+        return $this->config->get('food_voucher_api')['auth_token'];
+    }
+
+    private function userIsForced(): bool
+    {
+        $user = $this->auth->user();
+        return $user->state->force_active && $this->config->get('enable_force_active')
+            || $user->state->force_food && $this->config->get('enable_force_food');
+    }
+
+    /**
+     * @throws ErrorException
+     */
+    private function getInfo(bool $crew, ?array $userMealVouchers = null): array
+    {
+        $this->checkConfig();
+
+        $data = cache('foodVoucherInfo', function () {
```

Comment:
an explicit method would make testing easier but we dont have any tests here :zany_face:

Reply:
done